hpx/traits/is_executor_parameters.hpp

```diff
@@ -23,37 +23,83 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v3)
+    struct executor_parameters_mark_begin_end_tag : executor_parameters_tag {};
+
+    /// Parameters implement function std::size_t processing_units_count();
+    struct executor_parameters_processing_units_count_tag : executor_parameters_tag {};
+
+    /// Parameters implement function: void reset_thread_distribution(Executor &)
+    struct executor_parameters_reset_thread_distr_tag : executor_parameters_tag {};
 
     namespace detail
     {
         /// \cond NOINTERNAL
-        template <typename T>
+        template <typename T, typename Tag>
         struct is_executor_parameters
-          : std::is_base_of<executor_parameters_tag, T>
+          : std::is_base_of<Tag, T>
         {};
 
-        template <>
-        struct is_executor_parameters<executor_parameters_tag>
+        template <typename Tag>
+        struct is_executor_parameters<Tag, Tag>
           : std::false_type
         {};
 
-        template <typename T>
-        struct is_executor_parameters< ::boost::reference_wrapper<T> >
-          : is_executor_parameters<typename hpx::util::decay<T>::type>
+        template <typename T, typename Tag>
+        struct is_executor_parameters< ::boost::reference_wrapper<T>, Tag >
+          : is_executor_parameters<typename hpx::util::decay<T>::type, Tag>
         {};
 
 #if defined(HPX_HAVE_CXX11_STD_REFERENCE_WRAPPER)
-        template <typename T>
-        struct is_executor_parameters< ::std::reference_wrapper<T> >
-          : is_executor_parameters<typename hpx::util::decay<T>::type>
+        template <typename T, typename Tag>
+        struct is_executor_parameters< ::std::reference_wrapper<T>, Tag >
+          : is_executor_parameters<typename hpx::util::decay<T>::type, Tag>
         {};
 #endif
         /// \endcond
     }
 
     template <typename T>
     struct is_executor_parameters
-      : detail::is_executor_parameters<typename hpx::util::decay<T>::type>
+      : detail::is_executor_parameters<
+            typename hpx::util::decay<T>::type, executor_parameters_tag
+        >
+    {};
+
+    template <typename T>
+    struct is_executor_parameters_chunk_size
+      : detail::is_executor_parameters<
+            typename hpx::util::decay<T>::type, executor_parameters_chunk_size_tag
+        >
+    {};
+
+    template <typename T>
+    struct is_executor_parameters_mark_begin_end
+      : detail::is_executor_parameters<
+            typename hpx::util::decay<T>::type, executor_parameters_mark_begin_end_tag
+        >
+    {};
+
+    template <typename T>
+    struct is_executor_parameters_reset_thread_distr
+      : detail::is_executor_parameters<
+            typename hpx::util::decay<T>::type, executor_parameters_reset_thread_distr_tag
+        >
+    {};
+
+    template <typename T>
+    struct is_executor_parameters_processing_units_count
+      : detail::is_executor_parameters<
+            typename hpx::util::decay<T>::type,
+            executor_parameters_processing_units_count_tag
+        >
     {};
```

Comment:
If anyone is reading that and wondering: we have discussed it already with Hartmut outside GitHub. So far we haven't found better solution.